### PR TITLE
dev-util/gnome-builder: Bump to 46.3, gui-libs/libpanel: Bump to 1.6.0, dev-util/sysprof, dev-util/sysprof-capture, dev-util/sysprof-common: Bump to 46.0

### PR DESCRIPTION
**Pull request by flint2**

### CODE REVIEW

**Comment by PeakKS** [comment]
FYI builder 46.2 needs a newer version of VTE than what is currently packaged

**Comment by flint2** [comment]
> FYI builder 46.2 needs a newer version of VTE than what is currently packaged (see https://gitlab.gnome.org/GNOME/gnome-builder/-/commit/61b5a25bc4b36233420df8b1ee19bfae0e9ca2f0)

AFAIK dependency is 0.70.0
What version is needed and what version is "currently packaged"?
In gentoo master we have 0.74.2

From upstream:
```
meson.build:317 
libvte_dep = dependency('vte-2.91-gtk4', version: '>= 0.70.0')
```

**Comment by PeakKS** [comment]
> > FYI builder 46.2 needs a newer version of VTE than what is currently packaged (see [gitlab.gnome.org/GNOME/gnome-builder/-/commit/61b5a25bc4b36233420df8b1ee19bfae0e9ca2f0](https://gitlab.gnome.org/GNOME/gnome-builder/-/commit/61b5a25bc4b36233420df8b1ee19bfae0e9ca2f0))
> 
> AFAIK dependency is 0.70.0 What version is needed and what version is "currently packaged"? In gentoo master we have 0.74.2
> 
> From upstream:
> 
> ```
> meson.build:317 
> libvte_dep = dependency('vte-2.91-gtk4', version: '>= 0.70.0')
> ```

Yeah they did not update meson.build properly, it says 0.70.0 but that is outdated because of the commit I linked. That function was introduced to VTE here https://gitlab.gnome.org/GNOME/vte/-/commit/1ce1b836c0c002e0f841fc6cc3ea27337de8b319, and the first release that commit appears in looks to be 0.75.0.

**Comment by flint2** [comment]
> > > FYI builder 46.2 needs a newer version of VTE than what is currently packaged (see [gitlab.gnome.org/GNOME/gnome-builder/-/commit/61b5a25bc4b36233420df8b1ee19bfae0e9ca2f0](https://gitlab.gnome.org/GNOME/gnome-builder/-/commit/61b5a25bc4b36233420df8b1ee19bfae0e9ca2f0))
> > 
> > 
> > AFAIK dependency is 0.70.0 What version is needed and what version is "currently packaged"? In gentoo master we have 0.74.2
> > From upstream:
> > ```
> > meson.build:317 
> > libvte_dep = dependency('vte-2.91-gtk4', version: '>= 0.70.0')
> > ```
> 
> Yeah they did not update meson.build properly, it says 0.70.0 but that is outdated because of the commit I linked. That function was introduced to VTE here https://gitlab.gnome.org/GNOME/vte/-/commit/1ce1b836c0c002e0f841fc6cc3ea27337de8b319, and the first release that commit appears in looks to be 0.75.0.

Changed dep to 0.76.0, included in this draft PR https://github.com/gentoo/gentoo/pull/34844

It will be good to raise an issue and/or PR to upstream

File: gui-libs/libpanel/libpanel-1.6.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+EAPI=8
+
+inherit gnome.org meson vala xdg
+
+DESCRIPTION="A dock/panel library for GTK 4"
+HOMEPAGE="https://gitlab.gnome.org/GNOME/libpanel"
+
+LICENSE="LGPL-3+"
+SLOT="1"
+KEYWORDS="~amd64"
```
Comment (pacho2) [inline gui-libs/libpanel/libpanel-1.6.0.ebuild:13]: arm64 should be readded I think

File: dev-util/gnome-builder/gnome-builder-46.3.ebuild [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+
+EAPI=8
+PYTHON_COMPAT=( python3_{10..12} )
+DISABLE_AUTOFORMATTING=1
+FORCE_PRINT_ELOG=1
+
+inherit gnome.org gnome2-utils llvm meson optfeature python-single-r1 readme.gentoo-r1 virtualx xdg
+
+DESCRIPTION="An IDE for writing GNOME-based software"
+HOMEPAGE="https://wiki.gnome.org/Apps/Builder https://gitlab.gnome.org/GNOME/gnome-builder"
```
Comment (pacho2) [inline dev-util/gnome-builder/gnome-builder-46.3.ebuild:12]: Please use the newer web:
https://apps.gnome.org/Builder/

File: dev-util/gnome-builder/gnome-builder-46.3.ebuild [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+RDEPEND="
+	>=dev-libs/glib-2.75:2
+	>=gui-libs/gtk-4.10:4[introspection]
+	>=gui-libs/libadwaita-1.5_alpha:1
+	>=gui-libs/libpanel-1.5.0:1
+	>=gui-libs/gtksourceview-5.8:5[introspection]
+	>=dev-libs/json-glib-1.2.0
+	>=dev-libs/jsonrpc-glib-3.43.0:=
+	>=dev-libs/libdex-0.2:=
+	>=dev-libs/libpeas-1.99.0[python,${PYTHON_SINGLE_USEDEP}]
```
Comment (pacho2) [inline dev-util/gnome-builder/gnome-builder-46.3.ebuild:44]: Please set the right slot for libpeas

File: dev-util/gnome-builder/gnome-builder-46.3.ebuild [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+# desktop-file-utils required for tests, but we have it in deptree for xdg update-desktop-database anyway, so be explicit and unconditional
+# appstream-glib needed for validation with appstream-util with FEATURES=test
+BDEPEND="
+	doc? (
+		$(python_gen_cond_dep '
+			dev-python/sphinx[${PYTHON_USEDEP}]
+			dev-python/sphinx-rtd-theme[${PYTHON_USEDEP}]
+		')
+	)
+	gtk-doc? (
```
Comment (pacho2) [inline dev-util/gnome-builder/gnome-builder-46.3.ebuild:86]: shouldn't this be moved to "doc" USE flag? It seems it migrated away from gtk-doc

**Comment by pacho2** [comment]
Is sysprof-46 been handled in a different PR? I could find an older PR for 45 version but, at this stage, I was aiming at simply jumping to 46 if possible

Thanks a lot

File: dev-util/gnome-builder/gnome-builder-46.3.ebuild (hunk 4 above)
Comment (flint2) [reply dev-util/gnome-builder/gnome-builder-46.3.ebuild:86]: upstream has both doc and help 

meson_options.txt
```
option('help', type: 'boolean', value: false)
option('docs', type: 'boolean', value: false)
```

gnome-builder-46.3.ebuild
```
        $(meson_use doc help)
        $(meson_use gtk-doc docs)
```

What is the recommended approach for managing USE flags in this case?

File: dev-util/gnome-builder/gnome-builder-46.3.ebuild (hunk 2 above)
Comment (flint2) [reply dev-util/gnome-builder/gnome-builder-46.3.ebuild:12]: Done in https://github.com/gentoo/gentoo/pull/34771/commits/2e79d01ce9014f788953f80376df87ba8186e743

File: dev-util/gnome-builder/gnome-builder-46.3.ebuild (hunk 3 above)
Comment (flint2) [reply dev-util/gnome-builder/gnome-builder-46.3.ebuild:44]: Done in https://github.com/gentoo/gentoo/pull/34771/commits/2e79d01ce9014f788953f80376df87ba8186e743

**Comment by flint2** [comment]
> Is sysprof-46 been handled in a different PR? I could find an older PR for 45 version but, at this stage, I was aiming at simply jumping to 46 if possible
> 
> Thanks a lot

Added https://github.com/gentoo/gentoo/pull/34771/commits/33f085d5ab440e590383d0c5f952da8e37d71f15 https://github.com/gentoo/gentoo/pull/34771/commits/54231b2b2d19d10a723508b253ded743ef114b5e https://github.com/gentoo/gentoo/pull/34771/commits/deb742c72d7a8ec3a11102d1f5b8c44be9ae6997

File: dev-util/gnome-builder/gnome-builder-46.3.ebuild (hunk 4 above)
Comment (pacho2) [reply dev-util/gnome-builder/gnome-builder-46.3.ebuild:86]: You are right, probably as it is now (doc -> help, gtk-doc) is ok

**Pull request closed without merging**